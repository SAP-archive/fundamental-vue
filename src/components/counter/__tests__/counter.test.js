import { Counter } from '../'
import { mount } from '@vue/test-utils'

describe('Counter', () => {
  it('adds counter class and aria label and renders actual value', () => {
    const counter = mount(Counter, {
      propsData: {
        label: 'Counter',
        value: 44
      }
    })
    expect(counter.classes()).toContain('fd-counter')
    expect(counter.attributes('aria-label')).toEqual('Counter')
    expect(counter.text()).toEqual('44')
  })

  it('adds notification class when notification flag is used', () => {
    const counter = mount(Counter, {
      propsData: {
        label: 'Counter',
        notification: true
      }
    })
    expect(counter.classes()).toContain('fd-counter--notification')
  })

  it('formats value > 999 as 999+', () => {
    const counter = mount(Counter, {
      propsData: {
        label: 'Counter',
        value: 1200
      }
    })
    expect(counter.text()).toEqual('999+')
  })
})
