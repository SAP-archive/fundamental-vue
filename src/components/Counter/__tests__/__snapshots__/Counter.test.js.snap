// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter renders correctly 1`] = `
<span
  arialabel="Counter"
  class="fd-counter"
>
  44
</span>
`;

exports[`Counter renders correctly 2`] = `
<span
  class="fd-counter fd-counter--notification"
>
  0
</span>
`;
